server/src/main/java/org/apache/gravitino/server/web/rest/MetadataObjectPolicyOperations.java

```diff
@@ -0,0 +1,348 @@
+
+@Path("/metalakes/{metalake}/objects/{type}/{fullName}/policies")
+public class MetadataObjectPolicyOperations {
+  private static final Logger LOG = LoggerFactory.getLogger(MetadataObjectPolicyOperations.class);
+
+  private final PolicyDispatcher policyDispatcher;
+
+  @Context private HttpServletRequest httpRequest;
+
+  @Inject
+  public MetadataObjectPolicyOperations(PolicyDispatcher policyDispatcher) {
+    this.policyDispatcher = policyDispatcher;
+  }
+
+  @GET
+  @Path("{policy}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-object-policy." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-object-policy", absolute = true)
+  public Response getPolicyForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      @PathParam("policy") String policyName) {
+    LOG.info(
+        "Received get policy {} request for object type: {}, full name: {} under metalake: {}",
+        policyName,
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            MetadataObject object =
+                MetadataObjects.parse(
+                    fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+            Optional<Policy> policy = getPolicyForObject(metalake, object, policyName);
+
+            if (policy.isPresent()) {
+              LOG.info(
+                  "Get policy: {} for object type: {}, full name: {} under metalake: {}",
+                  policyName,
+                  type,
+                  fullName,
+                  metalake);
+              return Utils.ok(
+                  new PolicyResponse(DTOConverters.toDTO(policy.get(), Optional.of(false))));
+            }
+
+            // ensure the policy exists and is inheritable
+            Policy targetPolicy = policyDispatcher.getPolicy(metalake, policyName);
+            if (!targetPolicy.inheritable()
+                || !targetPolicy.supportedObjectTypes().contains(object.type())) {
+              return logNotFoundPolicy(metalake, policyName, type, fullName);
+            }
+
+            if (hasConflictExclusivePolicy(metalake, object, targetPolicy)) {
+              return logNotFoundPolicy(metalake, policyName, type, fullName);
+            }
+
+            MetadataObject parentObject = MetadataObjects.parent(object);
+            while (parentObject != null) {
+              if (!targetPolicy.supportedObjectTypes().contains(parentObject.type())) {
+                // If the parent object type is not supported by the target policy, we skip it.
+                parentObject = MetadataObjects.parent(parentObject);
+                continue;
+              }
+
+              Policy[] parentPolicies =
+                  policyDispatcher.listPolicyInfosForMetadataObject(metalake, parentObject);
```

Comment:
Why do you use "list" rather than "get" for policy?

Reply:
Because the same type of policy may have different names, so we should list all policies for the object first